### DISCUSSION TITLE
Image integration - srcset/high DPI screen optimization

### DISCUSSION THREAD

**Discussion by osdiab** [withastro/roadmap#299] [Proposal]
Hi, I tried out the Astro Image integration but the one thing really missing to me is that there doesn't seem to be a natural way to make an `Image` be rendered at several DPI levels to deal with high DPI screens (very common these days). So for example if I use `<Image width={150} src={...} />` and look at it on my Macbook, guaranteed the image will look really bad because Macbooks have high DPI screens. I would have expected the `Image` component to produce a `srcset` attribute with upscaled versions, or to have some option to do so, but doesn't look like that's there. Guess I could make it myself with the `Picture` component, but doesn't feel very intuitive or natural. Would Astro be interested in having this sort of functionality? Thanks!

**Comment by bnonn** [comment]
I have just run into the same issue. I have a simple image that just needs a 2x version for hi-DPI screens, and was a bit surprised to find that Astro's image integration doesn't do this. Given Astro's emphasis on performance, it seems like a no-brainer to add this functionality.
